src/Compilers/Core/Portable/Collections/ImmutableArrayDictionary`2+Builder.cs

```diff
@@ -0,0 +1,230 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.CodeAnalysis.Collections
+{
+    partial struct ImmutableArrayDictionary<TKey, TValue>
```

Comment:
>ImmutableArrayDictionary [](start = 19, length = 24)

I didn't understand why we need to introduce our own `Builder` type. I'd think we can just  use `Dictionary` as the mutable builder and provide a `ToImmutable()` extension to make the wrapper type.

Reply:
➡️ This pull request intentionally exposes the full API surface area.